gocql_bench/bench_marshal_test.go

```diff
@@ -0,0 +1,331 @@
+package gocql_bench
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+)
+
+// Benchmark marshalling and unmarshalling simple types
+func BenchmarkMarshalUnmarshalSimpleTypes(b *testing.B) {
+	b.Run("Int", func(b *testing.B) {
+		tType := gocql.NewNativeType(4, gocql.TypeInt, "")
+
+		var val int = 42
+		for i := 0; i < b.N; i++ {
+			// Marshal
+			marshaled, err := gocql.Marshal(tType, val)
+			if err != nil {
+				b.Fatal(err)
+			}
+
+			// Unmarshal
+			var unmarshaled int
+			err = gocql.Unmarshal(tType, marshaled, &unmarshaled)
+			if err != nil {
+				b.Fatal(err)
+			}
+		}
+	})
+
+	b.Run("Blob", func(b *testing.B) {
+		tType := gocql.NewNativeType(4, gocql.TypeBlob, "")
+
+		var val = []byte("test blob data")
```

Comment:
for blob and text we would want to test following cases:
1. Small:  +-100b
2. Medium: 1kb
3. Big: 1M
4. Huge: 10M

for text let's make it a json.
for blob a binary data.